device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -136,6 +130,21 @@
         this.state = IotHubConnectionStatus.DISCONNECTED;
     }
 
+    DeviceIO(String hostName, IotHubClientProtocol protocol, SSLContext sslContext, ProxySettings proxySettings, long sendPeriodInMilliseconds, long receivePeriodInMilliseconds)
+    {
+        this.sendPeriodInMilliseconds = sendPeriodInMilliseconds;
+        this.receivePeriodInMilliseconds = receivePeriodInMilliseconds;
+
+        this.state = IotHubConnectionStatus.DISCONNECTED;
```

Comment:
question on a slightly unrelated-to-multiplexing note - I recently found that if the .NET sdk is initially disconnected right after initialization, it reports a "Disconnected" state, even though it is actively retrying at this point. Does java behave as expected?

Reply:
The Java SDK only reports when the connection status changes, so it should work as expected in that case, yes